object_store/src/path.rs

```
@@ -124,7 +360,27 @@ impl CloudConverter {
     /// Creates a cloud storage location by joining this `ObjectStorePath`'s
     /// parts with `DELIMITER`
     pub fn convert(object_store_path: &ObjectStorePath) -> String {
-        object_store_path.parts.iter().map(|p| &p.0).join(DELIMITER)
+        match &object_store_path.inner {
+            PathRepresentation::RawCloud(path) => path.to_owned(),
+            PathRepresentation::RawPathBuf(_path) => {
+                todo!("convert");
```

I wonder if this should return an error or something other than panic'ing?